DQMOffline/MuonDPG/src/BaseTnPEfficiencyTask.cc

```diff
@@ -187,7 +188,9 @@ bool BaseTnPEfficiencyTask::hasTrigger(std::vector<int>& trigIndices,
   float dR2match = 999.;
   for (int trigIdx : trigIndices) {
     const std::vector<std::string> trigModuleLabels = m_hltConfig.moduleLabels(trigIdx);
-    const unsigned trigModuleIndex = trigModuleLabels.size() - 2;
+
+    const unsigned trigModuleIndex =
+        std::find(trigModuleLabels.begin(), trigModuleLabels.end(), "hltBoolEnd") - trigModuleLabels.begin() - 1;
```

Comment:
This will not work if `trigModuleLabels` is empty.
I imagine that there is something elsewehere that prevents it being empty by construction, is it correct?

Reply:
Yes, basically `trigModuleLabels` is not empty if there is at least a path matching IsoMu trigger (this is done [here](https://github.com/llunerti/cmssw/blob/from-CMSSW_12_4_0_pre3/DQMOffline/MuonDPG/src/BaseTnPEfficiencyTask.cc#L62) inside `dqmBeginRun`). If no matching path are found then `trigIndices` is empty and the loop inside `hasTrigger` won't run.